app/controllers/predicates/AuthorisedAction.scala

```diff
@@ -110,6 +112,8 @@ class AuthorisedAction @Inject()()(implicit val authConnector: AuthConnector,
   private[predicates] def agentAuthentication[A](block: User[A] => Future[Result], mtdItId: String)
                                                 (implicit request: Request[A], hc: HeaderCarrier): Future[Result] = {
 
+    val optionalSessionId: String = request.headers.get("sessionId").getOrElse("")
```

Comment:
Could rename `sessionId` because it's not an `Option`

Reply:
I have amended this in the next commit